geometric_kernels/kernels/base.py

```diff
@@ -30,14 +30,23 @@ def space(self) -> T:
         return self._space
 
     @abc.abstractmethod
-    def K(self, X, X2=None, **kwargs) -> B.Numeric:
+    def init_params_and_state(self):
```

Comment:
```suggestion
    def init_params_and_state(self) -> Tuple[Mapping[str, B.Numeric], Mapping[str, B.Numeric]]:
```

Reply:
Apparently this doesn't work well with the typechecker.

> TypeError: Parameters to generic types must be types. Got lab.types.Numeric.